plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/file/FileMetastoreTableOperations.java

```diff
@@ -76,7 +80,13 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
             metastore.replaceTable(database, tableName, table, privileges);
         }
         catch (RuntimeException e) {
-            throw new CommitFailedException(e, "Failed to commit transaction to FileHiveMetastore");
+            if (e instanceof TrinoException trinoException &&
+                    trinoException.getErrorCode() == HIVE_METASTORE_ERROR.toErrorCode() &&
+                    Objects.equals(trinoException.getMessage(), "Cannot update Iceberg table: supplied previous location does not match current location")) {
+                // CommitFailedException is handled as a special case in the Iceberg library. This commit will automatically retry
+                throw new CommitFailedException(e, "Failed to replace table due to concurrent updates: %s.%s", database, tableName);
```

Comment:
nit: since this string constant is present in a couple of places in the codebase, is it worth pulling it out?